devnet-sdk/descriptors/chain_config.go

```diff
@@ -0,0 +1,181 @@
+package descriptors
+
+import (
+	"math/big"
+
+	"github.com/ethereum-optimism/optimism/op-node/rollup"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+type ChainConfig struct {
+	InteropTime  *uint64                `json:"interopTime,omitempty"` // Interop switch time (nil = no fork, 0 = already on optimism interop)
+	Optimism     *params.OptimismConfig `json:"optimism,omitempty"`
+	BedrockBlock *big.Int               `json:"bedrockBlock,omitempty"` // Bedrock switch block (nil = no fork, 0 = already on optimism bedrock)
+	RegolithTime *uint64                `json:"regolithTime,omitempty"` // Regolith switch time (nil = no fork, 0 = already on optimism regolith)
+	CanyonTime   *uint64                `json:"canyonTime,omitempty"`   // Canyon switch time (nil = no fork, 0 = already on optimism canyon)
+	// Delta: the*Delta upgrade does not affect the execution-layer, and is thus not configurable in the chain config.
+	EcotoneTime  *uint64 `json:"ecotoneTime,omitempty"`  // Ecotone switch time (nil = no fork, 0 = already on optimism ecotone)
+	FjordTime    *uint64 `json:"fjordTime,omitempty"`    // Fjord switch time (nil = no fork, 0 = already on Optimism Fjord)
+	GraniteTime  *uint64 `json:"graniteTime,omitempty"`  // Granite switch time (nil = no fork, 0 = already on Optimism Granite)
+	HoloceneTime *uint64 `json:"holoceneTime,omitempty"` // Holocene switch time (nil = no fork, 0 = already on Optimism Holocene)
+	IsthmusTime  *uint64 `json:"isthmusTime,omitempty"`  // Isthmus switch time (nil = no fork, 0 = already on Optimism Isthmus)
+}
+
+// IsForkActivated checks if a specific fork is activated at the given timestamp
+// based on the chain configuration.
+func (c *ChainConfig) IsForkActivated(forkName rollup.ForkName, timestamp uint64) bool {
+	switch forkName {
+	case rollup.Bedrock:
+		// Bedrock is activated based on block number, not timestamp
+		return true // Assuming bedrock is always active in the context of this validator
+	case rollup.Regolith:
+		return c.IsOptimismRegolith(timestamp)
+	case rollup.Canyon:
+		return c.IsOptimismCanyon(timestamp)
+	case rollup.Ecotone:
+		return c.IsOptimismEcotone(timestamp)
+	case rollup.Fjord:
+		return c.IsOptimismFjord(timestamp)
+	case rollup.Granite:
+		return c.IsOptimismGranite(timestamp)
+	case rollup.Holocene:
+		return c.IsOptimismHolocene(timestamp)
+	case rollup.Isthmus:
+		return c.IsOptimismIsthmus(timestamp)
+	case rollup.Interop:
+		return c.IsInterop(timestamp)
+	default:
+		return false
+	}
+}
+
+// CurrentFork returns the most recent fork that is currently activated.
+func (c *ChainConfig) CurrentFork(timestamp uint64) rollup.ForkName {
+	if c.IsOptimismIsthmus(timestamp) {
```

Comment:
We are missing interop here

Reply:
Interop doesn't belong here, it's not a normal fork.

Reply:
Removing `CurrentFork()`.